program/src/processor.rs

```diff
@@ -76,30 +76,38 @@ fn get_total_rewards_checked(
         return Err(PaladinRewardsError::IncorrectHolderRewardsPoolAddress.into());
     }
 
-    let holder_rewards_pool_data = holder_rewards_pool_info.try_borrow_data()?;
-    let holder_rewards_pool_state =
-        bytemuck::try_from_bytes::<HolderRewardsPool>(&holder_rewards_pool_data)
-            .map_err(|_| ProgramError::InvalidAccountData)?;
-
-    Ok(holder_rewards_pool_state.total_rewards)
+    Ok(())
 }
 
-fn calculate_reward_share(
-    token_supply: u64,
-    token_account_balance: u64,
-    total_rewards: u64,
-) -> Result<u64, ProgramError> {
+fn calculate_rewards_per_token(rewards: u64, token_supply: u64) -> Result<u128, ProgramError> {
     if token_supply == 0 {
         return Ok(0);
     }
-    // Calculation: (token_amount / total_token_supply) * pool_rewards
+    // Calculation: rewards / token_supply
     //
-    // However, multiplication is done first to avoid truncation, ie:
-    // (token_amount * pool_rewards) / total_token_supply
-    (token_account_balance as u128)
-        .checked_mul(total_rewards as u128)
+    // Scaled by 1e9 to store 9 decimal places of precision.
+    (rewards as u128)
+        .checked_mul(1_000_000_000)
```

Comment:
nit: Since the scaling factor is used in a couple of places, can you factor it into a `const REWARDS_PER_TOKEN_SCALING_FACTOR` or some better name?

Reply:
This name makes the most sense to me, tbh.